File: front/src/app/services/projet.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { Observable, Observer } from 'rxjs';
import { Projet} from '../models/projet';

@Injectable({
  providedIn: 'root'
})
export class ProjetService {

  private ProjetUrl: string = 'http://localhost:8080/api/projet';
  constructor(private http: HttpClient) { }

findAll(): Observable<Projet[]>{  
  return this.http.get<Projet[]>(this.ProjetUrl);
}

findById(id :number): Observable<Projet>{
  return this.http.get<Projet>(this.ProjetUrl + "/" + id);
}

save(projet: Projet): Observable<Projet>{
  console.log("b "+this.ProjetUrl);
  return this.http.post<Projet>(this.ProjetUrl, projet);
}

delete(id: number): Observable<Projet>{
  return this.http.get<Projet>(`${this.ProjetUrl}/${id}/delete`);
}

}

File: back/src/main/java/com/springboot/projetfinal/services/ProjetService.java
package com.springboot.projetfinal.services;

import java.util.List;

import com.springboot.projetfinal.model.Projet;


public interface ProjetService {

	public List<Projet> findAll();
	
	public Projet findById(Long id);
	
	public Projet save(Projet projet);
	
	public Projet deleteById(Long id);
	
}
